datafusion/core/src/physical_plan/aggregates/row_hash.rs

```diff
@@ -514,6 +580,19 @@ impl GroupedHashAggregateStream {
     }
 }
 
+fn get_accumulator_set_size(
+    groups_with_rows: &[usize],
+    row_group_states: &[RowGroupState],
+) -> usize {
+    groups_with_rows.iter().fold(0usize, |acc, group_idx| {
+        let group_state = &row_group_states[*group_idx];
+        group_state
+            .accumulator_set
+            .iter()
+            .fold(acc, |acc, accumulator| acc + accumulator.size())
+    })
+}
+
 /// The state that is built for each output group.
```

Comment:
I think the logic is quite complex to collect the memory size of the accumulators, maybe the computation is more than the real useful aggregations

Reply:
Yes, I agree the calculation for the memory size is complicated. 

I think @tustvold  has been thinking about how to improve performance in this area, but I am not sure how far he has gotten

In general, managing individual allocations (and then accounting for their sizes) for each group is a significant additional overhead for grouping.